core/src/main/scala/dagr/core/execsystem/TaskManagerLike.scala

```diff
@@ -0,0 +1,177 @@
+
+  /** Gets the task for the given id.  If there is no task being tracked
+    * with that id, throws a [[NoSuchElementException]].
+    *
+    * @param id the id to lookup.
+    * @return the associated task.
+    */
+  def apply(id: BigInt): Task
+
+  /** Gets the task execution info for the given task.  If the task is not
+    *  being tracked, throws a [[NoSuchElementException]].
+    *
+    * @param task the task to lookup.
+    * @return the associated task execution info.
+    */
+  def apply(task: Task): BaseInfoAndNode
+
+  /** Get the bi-directional map between managed tasks and their associated task execution information.
+    *
+    * @return the task and task execution information bi-directional map.
+    */
+  def taskToInfoBiMapFor: BiMap[Task, TaskExecutionInfo]
+
+  /** Checks if we have failed tasks.
+    *
+    * @return true if we have failed tasks, false otherwise.
+    */
+  def hasFailedTasks: Boolean
+
+  /** Adds a task to be managed
+    *
+    * @param task the given task.
+    * @return the task identifier.
+    */
+  def addTask(task: Task): BigInt
+
+  /** Adds tasks to be managed
+    *
+    * @param tasks the given tasks.
+    * @return the task identifiers.
+    */
+  def addTasks(tasks: Task*): List[BigInt]
+
+  /** Resubmit a task for execution.  This will stop the task if it is currently running, and queue
+    * it up for execution.  The number of attempts will be reset to zero.
+    *
+    * @param task the task to resubmit.
+    * @return true if the task was successfully resubmitted, false otherwise.
+    */
+  // Turning it off until `resubmit` is used.
+  //def resubmitTask(task: Task): Boolean
+
+  /** Replace the original task with the replacement task and update any internal references.
+    *
+    * If the replacement task depends on any task, any tasks depends on the replacement task, or if the original
+    * task is not being tracked this method will return false.  Any tasks dependent on the original task, or any
+    * task on which the original task depends, will have their dependency relationship updated.
+    *
+    * @param original the original task to replace.
+    * @param replacement the replacement task.
+    * @return true if we are successful, false otherwise.
+    */
+  def replaceTask(original: Task, replacement: Task): Boolean
+
+  /** Run all tasks managed to either completion, failure, or inability to schedule.
+    *
+    * This will terminate tasks that were still running before returning in the case of failure or
+    * inability to schedule.
+    *
+    * @param sleepMilliseconds the time to wait in milliseconds to wait between trying to schedule tasks.
+    * @param timeout           the length of time in milliseconds to wait for running tasks to complete
+    * @return a bi-directional map from the set of tasks to their execution information.
+    */
+  def runToCompletion(sleepMilliseconds: Int = 1000, timeout: Int = 1000): BiMap[Task, TaskExecutionInfo]
```

Comment:
Could `timeout` be either a `java.util.concurrent.TimeUnit` or a `scala.concurrent.duration.Duration`?  It might also be nice to make it an `Option` type so you can specify no timeout if you desire?